syntax = "proto3";
package Client;

message ClientCommand {
    bytes UUID = 1;
    oneof Command {
        string log_message = 2;
        Result result = 3;
        Image image = 4;
        ControlChanged control_changed = 5;
    }
}

message Result {
    oneof Result {
        string success = 1;
        string failure = 2;
    }
}

message Image {
    bytes data = 1;
}

message ControlChanged {
    oneof Value {
        double number = 1;
        string text = 2;
        bool boolean = 3;
    }
}